components/raftstore/src/store/fsm/peer.rs

```diff
@@ -2186,6 +2186,23 @@ where
         if is_leader {
             self.fsm.peer.approximate_size = estimated_size;
             self.fsm.peer.approximate_keys = estimated_keys;
+
+            if let Some(estimated_size) = self.fsm.peer.approximate_size {
+                if estimated_size > self.ctx.coprocessor_host.cfg.region_max_size.0 {
+                    info!(
+                        "trigger split check immediately";
+                        "region_id" => self.fsm.region_id(),
+                        "peer_id" => self.fsm.peer_id(),
+                        "size" => estimated_size,
+                    );
+                    self.fsm.peer.size_diff_hint = self.ctx.cfg.region_split_check_diff.0;
+                    // trigger a split check immediately when size is still large
+                    self.ctx
```

Comment:
Seems duplicated with split tick scheduled at the beginning of the function.

Reply:
that will be scheduled after 10s, but this is intended to trigger split check at once.

Reply:
Then the range will be scan twice.

Reply:
The `size_diff_hint` will let the second split check skip. But it's indeed better to not schedule at the beginning if this condition is hit, will do.